src/Tokenizer.js

```diff
@@ -479,7 +495,16 @@ module.exports = class Tokenizer {
       } else {
         title = cap[3] ? cap[3].slice(1, -1) : '';
       }
-      href = href.trim().replace(/^<([\s\S]*)>$/, '$1');
+
+      href = href.trim();
+      if (href.startsWith('<')) {
+        if (this.options.pedantic && !trimmedUrl.endsWith('>')) {
+          // pedantic allows starting angle bracket without ending angle bracket
+          href = href.slice(1);
+        } else {
+          href = href.slice(1, -1);
+        }
+      }
```

Comment:
I liked the original way of doing this with the regex. My impression is regex is generally faster than manually checking characters (but I may be wrong).

Could the original regex be updated to this? (just add a `?` so it allows a missing end `>`):
`href = href.trim().replace(/^<([\s\S]*)>?$/, '$1');`

Reply:
regex is faster if there isn't [catastrophic backtracking](https://regular-expressions.mobi/catastrophic.html?wlr=1) that can allow ReDOS attacks. The problem with just using regex is that the end bracket can be escaped and as mentioned above that is vulnerable to ReDOS.

Reply:
Hmm... I agree with the concern, but I'm not sure if it applies to this pattern. How does this lead to catastrophic backtracking? There is no nested repeating element in this regex, and no branching paths. I tested this just for fun on regex101.com, and even with thousands of characters it completes the match (or non-match) in 8 steps. I don't think this pattern *can* backtrack because there is only one path it can take from the `^` to the `$`. There is not any alternate `|` and all of the capturing takes place in a single, non-repeated group.

Reply:
Meh, I missed your comment about the end bracket not being escaped, and that leads to the same problem with the regex above. Alright, non-regex it is. 👍